chisel/src/main.rs

```diff
@@ -6,65 +6,102 @@ mod driver;
 mod options;
 mod result;
 
-use clap::{crate_description, crate_name, crate_version};
-use std::fs::{read, read_to_string, write};
 use std::process;
 
-use clap::{App, Arg, ArgMatches, SubCommand};
-use serde_yaml::Value;
+use clap::{crate_description, crate_name, crate_version, App, Arg, SubCommand};
 
-use libchisel::{
-    checkstartfunc::*, deployer::*, dropsection::*, remapimports::*, remapstart::*, repack::*,
-    snip::*, trimexports::*, trimstartfunc::*, verifyexports::*, verifyimports::*,
-};
+use cmd_run::chisel_run;
+use options::ChiselFlags;
 
 fn fail(code: i32, message: &str) -> ! {
     eprintln!("{}: {}", crate_name!(), message);
     process::exit(code);
 }
 
-#[cfg(feature = "binaryen")]
-use libchisel::binaryenopt::*;
-
-use cmd_run::chisel_subcommand_run;
-
-static ERR_NO_SUBCOMMAND: &'static str = "No subcommand provided.";
-
-fn err_exit(msg: &str) -> ! {
-    eprintln!("{}: {}", crate_name!(), msg);
-    process::exit(-1);
-}
-
 pub fn main() {
     let cli_matches = App::new("chisel")
         .version(crate_version!())
         .about(crate_description!())
         .arg(
-            Arg::with_name("VERBOSE")
-                .short("v")
-                .long("verbose")
-                .help("Enables verbose debug logging")
+            Arg::with_name("NO_RECOVER")
+                .short("x")
+                .long("norecover")
+                .help("Exits immediately on all recoverable errors")
                 .global(true),
         )
+        .arg(
+            Arg::with_name("DEBUG_MESSAGES")
+                .short("d")
+                .long("debug")
+                .help("Enables debug messages")
+                .global(true),
+        )
+        .arg(
+            Arg::with_name("MODULES")
+                .short("m")
+                .long("modules")
+                .takes_value(true)
+                .multiple(true)
+                .help("Selects modules to use"),
+        )
+        .arg(Arg::with_name("FILE").help("File to chisel"))
```

Comment:
This doesn't have a short or long option, does this mean it is a leaf argument?

Reply:
If so perhaps it should be last before the help message.

Reply:
It's an argument which is invoked without a flag. Will move to end.